src/servermap.cpp

```diff
@@ -261,6 +262,20 @@ bool ServerMap::initBlockMake(v3s16 blockpos, BlockMakeData *data)
 	return true;
 }
 
+void ServerMap::cancelBlockMake(BlockMakeData *data)
+{
+	assert(data->vmanip); // no vmanip = initBlockMake did not complete (caller mistake)
+	v3s16 bpmin = data->blockpos_min;
```

Comment:
Shouldn't `v3s16 extra_borders(1, 1, 1);` (possibly as `constexpr static` variable) be used here? Otherwise it is not clear whether there's always one `refDrop` for each `refGrab`.

Reply:
Comment from repetitivestrain: i think the relationship between extra_borders and the loops in finishBlockMake and cancelBlockMake is quite manifest